assets/src/edit-story/components/panels/size.js

```diff
@@ -18,101 +18,218 @@
+              ...state,
+              y: isNaN(value) || value === '' ? '' : parseFloat(value),
+            })
+          }
+          disabled={isFill}
+        />
+      </Row>
+      {/** Width/height & lock ratio */}
+      <Row expand>
+        <BoxedNumeric
+          prefix={_x('W', 'The Width dimension', 'web-stories')}
+          value={state.width}
+          isMultiple={width === ''}
+          onChange={(value) => {
+            const ratio = width / height;
+            const newWidth =
+              isNaN(value) || value === '' ? '' : parseFloat(value);
+            setState({
+              ...state,
+              width: newWidth,
+              height:
+                height !== '' && typeof newWidth === 'number' && lockRatio
+                  ? dataPixels(newWidth / ratio)
+                  : height,
+            });
+          }}
+          disabled={isFill}
+        />
+        <Toggle
+          icon={<Locked />}
+          uncheckedIcon={<Unlocked />}
+          value={lockRatio}
+          isMultiple={false}
+          onChange={(value) => {
+            setLockRatio(value);
+          }}
+          disabled={isFill}
+        />
+        <BoxedNumeric
+          prefix={_x('H', 'The Height dimension', 'web-stories')}
+          value={state.height}
+          isMultiple={height === ''}
+          onChange={(value) => {
+            const ratio = width / height;
+            const newHeight =
+              isNaN(value) || value === '' ? '' : parseFloat(value);
+            setState({
+              ...state,
+              height: newHeight,
+              width:
+                width !== '' && typeof newHeight === 'number' && lockRatio
+                  ? dataPixels(newHeight * ratio)
+                  : width,
+            });
+          }}
+          disabled={isFill}
+        />
+      </Row>
+      {/** Fill & Reset size */}
+      <Row expand={false} spaceBetween={false}>
+        <Toggle
+          icon={<Fullbleed />}
+          value={state.isFill}
+          isMultiple={false}
+          onChange={(value) => {
+            setState({
+              ...state,
+              isFill: value,
+            });
+          }}
+        />
+        {/** TODO: Implement size resetting */}
+        <Button onClick={() => {}}>{__('Reset size', 'web-stories')}</Button>
+      </Row>
+      {/** Rotation and Flipping */}
+      <Row expand={false} spaceBetween={false}>
+        <ExpandedNumeric
+          label={__('Rotate', 'web-stories')}
+          suffix={_x('°', 'Degrees, 0 - 360. ', 'web-stories')}
+          value={state.rotationAngle}
+          isMultiple={rotationAngle === ''}
+          onChange={(value) => {
+            setState({
+              ...state,
+              rotationAngle:
+                isNaN(value) || value === '' ? '' : parseFloat(value),
+            });
+          }}
+          disabled={isFill}
+        />
+        <FlipButton>
+          <FlipButtonIcon as={FlipHorizontal} />
```

Comment:
What? Can you use `as` for this? That's pretty cool! TIL

Reply:
I checked this part and it's not working as expected. So I updated this parts.